apps/oc_chef_wm/src/oc_chef_wm_named_policy.erl

```diff
@@ -0,0 +1,205 @@
+            {keys, {string_match, chef_regex:regex_for(cookbook_name)}},
+            {values,
+             %% apparently there's no cleaner way to do that with `ej:valid' (??)
+             {fun_match, {fun valid_cookbook_lock/1, object,
+                          <<"Invalid cookbook_lock constraint">>}}}}}}
+          ]}).
+
+-define(COOKBOOK_LOCK_VAIDATION_CONSTRAINTS,
+    {[{<<"dotted_decimal_identifier">>,
+       chef_cookbook_version:single_cookbook_version_spec()},
+      {<<"identifier">>,
+       {string_match, chef_regex:regex_for(policy_identifier)}},
+
+      {{opt, <<"version">>},
+       chef_cookbook_version:single_cookbook_version_spec()}]}).
+
+valid_cookbook_lock(CookbookLockJson) ->
+    case ej:valid(?COOKBOOK_LOCK_VAIDATION_CONSTRAINTS, CookbookLockJson) of
+        ok -> ok;
+        _Bad -> error
+    end.
+
+init(Config) ->
+    oc_chef_wm_base:init(?MODULE, Config).
+
+init_resource_state(_Config) ->
+    {ok, #policy_state{}}.
+
+request_type() ->
+    "policies".
+
+allowed_methods(Req, State) ->
+    {['GET', 'PUT', 'POST', 'DELETE'], Req, State}.
+
+post_is_create(Req, State) ->
+    {true, Req, State}.
+
+create_path(Req, State) ->
+    Name = wrq:path_info(policy_name, Req),
+    {Name, Req, State}.
+
+validate_request(Method, Req, State = #base_state{organization_guid = OrgId}) when Method == 'GET';
+                                          Method == 'DELETE' ->
+    Name = wrq:path_info(policy_name, Req),
+    Group = wrq:path_info(policy_group, Req),
+    {Req, State#base_state{superuser_bypasses_checks = true, resource_state = #policy_state{oc_chef_policy = #oc_chef_policy{org_id = OrgId, name = Name, policy_group = Group}}}};
+validate_request(_PostOrPut, Req, #base_state{organization_guid = OrgId, resource_state = PolicyState} = State) ->
+    Policy = validate_json(Req),
+    Group = wrq:path_info(policy_group, Req),
+    NewPolicy = ej:set({<<"policy_group">>}, Policy, erlang:list_to_binary(Group)),
+    {Req, State#base_state{
+            superuser_bypasses_checks = true,
+            resource_state = PolicyState#policy_state{
+                               oc_chef_policy = #oc_chef_policy{org_id = OrgId, policy_group = Group},
+                               policy_data = NewPolicy}}}.
+
+validate_json(Req) ->
+    Body = wrq:req_body(Req),
+    {ok, Policy} = oc_chef_policy:parse_binary_json(Body),
+    ok = validate_name(Req, Policy),
+    case ej:valid(?VALIDATION_CONSTRAINTS, Policy) of
+        ok ->
+            Policy;
+        Bad ->
+            throw(Bad)
+    end.
+
+validate_name(Req, Policy) ->
+    NameFromReq = wrq:path_info(policy_name, Req),
+    NameFromJson = erlang:binary_to_list(ej:get({<<"name">>}, Policy)),
+    case ibrowse_lib:url_encode(NameFromJson) =:= NameFromReq of
+        true ->
+            ok;
+        false ->
+            erlang:throw({error, {mismatch, {<<"Name">>, ignored, ignored}}})
+    end.
+
+auth_info(Req, #base_state{chef_db_context = DbContext,
```

Comment:
Minor nitpick but  I'd rather see this near the top of the module - it's called earlier in the request, and having it there makes it easy [relatively] to see permissions requirements without searching.